source/test/TestApp/PlayFabEventTest.cpp

```diff
@@ -218,6 +229,40 @@ namespace PlayFabUnit
         std::bind(&PlayFabEventTest::NonStaticEmitEventCallback, this, std::placeholders::_1, std::placeholders::_2));
     }
 
+    void PlayFabEventTest::GenericMultiThreadedTest(unsigned int pNumThreads, unsigned int pNumEventsPerThread)
+    {
+        numEventsHeard = 0;
+        for (unsigned int thread = 0; thread < pNumThreads; ++thread)
+        {
+            testThreadPool.emplace_back(std::thread([pNumEventsPerThread, this](){
```

Comment:
>emplace_back(std::thread [](start = 27, length = 24)

the whole point of emplace_back is to pass the args that are forwarded to the constructor so it doesn't make sense to call emplace_back with an already constructed value.

e.g.
```
std::vector<Type> x;
x.emplace_back( X(arg1, arg2, arg3) ); // wrong/unnecessary
x.emplace_back(arg1, arg2, arg3); // correct use of emplace_back
x.push_back(X(arg1, arg2, arg3)); // if you want the value passed into the vector to explicitly have the type construction somewhere, push_back would be the method to use
```

Reply:
Or stick with emplace_back but skip the thread ctor as follows:

testThreadPool.emplace_back([pNumEventsPerThread, this]() { ... });

---
In reply to: [345433235](https://github.com/PlayFab/XPlatCoreTemplate/pull/44#discussion_r345433235) [](ancestors = 345433235)